### PR TITLE
[Enhancement](chore) Some Status relevant enhancement

### PR DESCRIPTION
**Pull request by platoneko**
## Proposed changes

Issue Number: close #xxx

- Remove reduntant `Status::is_xxx` methods
- Make `capture_stacktrace` consteval
- Fix some abuse of status comparison

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/http/action/download_action.cpp [hunk 1]
```diff
@@ -68,7 +69,7 @@ void DownloadAction::handle_normal(HttpRequest* req, const std::string& file_par
         status = check_token(req);
         if (!status.ok()) {
             std::string error_msg = status.to_string();
-            if (status.is_not_authorized()) {
```
Comment (yiguolei) [inline be/src/http/action/download_action.cpp:71]: we should remove these method:
    bool is_invalid_argument() const { return ErrorCode::INVALID_ARGUMENT == _code; }

    bool is_not_found() const { return _code == ErrorCode::NOT_FOUND; }
    bool is_not_authorized() const { return code() == TStatusCode::NOT_AUTHORIZED; }

I think only IO ERROR should has method is_io_error.

File: be/src/common/status.h [hunk 2]
```diff
@@ -413,17 +424,6 @@ class Status {
 
     bool ok() const { return _code == ErrorCode::OK; }
 
-    bool is_io_error() const {
```
Comment (yiguolei) [inline be/src/common/status.h:416]: are you sure we could remove is_io_error method?

File: be/src/common/status.h [hunk 3]
```diff
@@ -346,10 +343,24 @@ class Status {
...
+        if constexpr (stacktrace && capture_stacktrace(code)) {
+            status._err_msg->_stack = get_stack_trace();
+            LOG(WARNING) << "meet error status: " << status; // may print too many stacks.
+        }
+#endif
+        return status;
     }
 
-    template <bool stacktrace = true, typename... Args>
+    template <bool stacktrace = false, typename... Args>
```
Comment (yiguolei) [inline be/src/common/status.h:363]: why change the value to false?

File: be/src/common/status.h [hunk 4]
```diff
@@ -346,10 +343,24 @@ class Status {
...
     Status static Error(int code, std::string_view msg, Args&&... args) {
```
Comment (yiguolei) [inline be/src/common/status.h:364]: this method has many duplicate codes with the previous method.

File: be/src/common/status.h (hunk 2 above)
Comment (platoneko) [reply be/src/common/status.h:416]: I have checked these error code in `is_io_error`, replace `IO_ERROR` to `TEST_FILE_ERROR`; removed `READ_UNENOUGH` as this error code is never used; replace `CHECKSUM_ERROR` and `FILE_DATA_ERROR` to `INTERNAL_ERROR` as such 2 error code are only used in meta header file.

File: be/src/common/status.h [hunk 5]
```diff
@@ -276,7 +273,7 @@ E(INVERTED_INDEX_BUILD_WAITTING, -6008);
 
 // clang-format off
 // whether to capture stacktrace
-inline bool capture_stacktrace(int code) {
+consteval bool capture_stacktrace(int code) {
     return code != ErrorCode::OK
         && code != ErrorCode::END_OF_FILE
```
Comment (yiguolei) [inline be/src/common/status.h:278]: Status check if could segment compaction() {
return not_ready();
}

**Comment by yiguolei** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.53 seconds
 stream load tsv:          537 seconds loaded 74807831229 Bytes, about 132 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.6 seconds inserted 10000000 Rows, about 337K ops/s
 storage size: 17162084363 Bytes

**Comment by yiguolei** [comment]
run buildall

**Review by yiguolei — APPROVED** [review]
LGTM

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 48.35 seconds
 stream load tsv:          553 seconds loaded 74807831229 Bytes, about 129 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          66 seconds loaded 1101869774 Bytes, about 15 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 insert into select:          29.5 seconds inserted 10000000 Rows, about 338K ops/s
 storage size: 17162060247 Bytes

**Review by gavinchou — APPROVED** [review]
LGTM

**Pull request merged**